pkg/tasks/thanos_ruler_user_workload.go

```diff
@@ -41,7 +43,7 @@ func (t *ThanosRulerUserWorkloadTask) Run(ctx context.Context) error {
 		return t.create(ctx)
 	}
 
-	return t.destroy(ctx)
+	return t.cleanup(ctx)
```

Comment:
If we combine cleanup and destroy, we'll only ever cleanup the obsolete resources when a user removes their UWM stack.
Maybe its better to add a `cleanup()` function and call it at the end of both `create()` and `destroy()`?

Reply:
Nice catch, I missed to do that in user_workload but done as separate method in others. Updating same way